src/Shared/XMakeElements.cs

```diff
@@ -35,25 +35,25 @@ internal static class XMakeElements
         internal const string usingTaskBody = "Task";
         internal const string sdk = "Sdk";
 
-        internal static readonly char[] InvalidTargetNameCharacters = { '$', '@', '(', ')', '%', '*', '?', '.' };
+        internal static char[] InvalidTargetNameCharacters => [ '$', '@', '(', ')', '%', '*', '?', '.' ];
```

Comment:
I would prefer these two to continue being `ReadOnly`.

Reply:
Good catch @maridematte!

@kasperk81 Look at https://sharplab.io/#v2:CYLg1APgAgTAjAWAFBQMwAJboMLoN7LpGYZRwBs6ATgKYCGwA9gHYA2AnugMYAWdVAbQC66ACo8AlswDmAdQkAXHgFEAjgFc6rAM7oAvOgHoA5ABJjAGhNQolkwAo7xgJROApE4BUTgPxOAdMboQgDchMRomBTcfIIi4lJyijwAglRUjADu+gB8hibmVsY2To5FrkUeRd5FfkWBwWFIAL5AA. See how the arrow mechanism allocates a new array on every call?

Reply:
This isn't a ROS though, it's an array.